openedx/features/survey_report/static/survey_report/js/admin_banner.js

```diff
@@ -4,8 +4,26 @@ $(document).ready(function(){
       // If you want to do something after the slide-up, do it here.
       // For example, you can hide the entire div:
       // $(this).hide();
+      var userId = document.getElementById('userIdSurvey').value;
+      // Store the dismissal time in milliseconds
+      var dismissalTime = new Date().getTime();
+      // Calculate the expiration time (1 month in milliseconds)
+      var expirationTime = dismissalTime + (30 * 24 * 60 * 60 * 1000); // 30 days
+      // Store the dismissal time in the browser's local storage
+      localStorage.setItem('bannerDismissalTime_' + userId, dismissalTime);
+      localStorage.setItem('bannerExpirationTime_' + userId, expirationTime);
     });
   });
+
+    // Check if the banner should be shown or hidden on page load
+  var userId = document.getElementById('userIdSurvey').value;
+  var bannerDismissalTime = localStorage.getItem('bannerDismissalTime_' + userId);
+  var bannerExpirationTime = localStorage.getItem('bannerExpirationTime_' + userId);
+  var currentTime = new Date().getTime();
+  if (bannerDismissalTime && bannerExpirationTime && currentTime < bannerExpirationTime) {
+    // Banner was dismissed and it's still within the expiration period, so hide it
+    $('#originalContent').hide();
```

Comment:
I tested the functionality and although it does hide the survey when I press the Dismiss button, every time I reload the page it shows and hides it quickly (milliseconds). Is it possible that the survey is never shown?

Reply:
To do that we would have to add the display: none property on the banner and toggle it with javascript as well when the document loads. It would have the same effect when the banner should appear.

Reply:
Perfect, it would be nice if it could be done.